cdc/model/changefeed.go

```diff
@@ -94,16 +94,14 @@ func (s FeedState) IsNeeded(need string) bool {
 }
 
 const (
-	// errorHistoryGCInterval represents how long we keep error record in changefeed info
-	errorHistoryGCInterval = time.Minute * 10
-
-	// errorHistoryCheckInterval represents time window for failure check
-	errorHistoryCheckInterval = time.Minute * 2
-
 	// ErrorHistoryThreshold represents failure upper limit in time window.
 	// Before a changefeed is initialized, check the the failure count of this
 	// changefeed, if it is less than ErrorHistoryThreshold, then initialize it.
-	ErrorHistoryThreshold = 3
+	ErrorHistoryThreshold = 1
+
+	// deltaBackoffInterval is added to the time window to safely ensure that
```

Comment:
Can you explain more why `deltaBackoffInterval` is needed? What would happen if we do not have it?

Reply:
> Can you explain more why `deltaBackoffInterval` is needed? What would happen if we do not have it?

If we don't have this `deltaBackoffInterval`, the `ErrorsReachedThreshold` will immediately return false on next tick because the changefeed will not return an error in such short period tick. In other words, this 30s delta interval is used to allow changefeed have a chance to run again and return an error.

Reply:
> Can you explain more why `deltaBackoffInterval` is needed? What would happen if we do not have it?

If we don't have `deltaBackoffInterval`, the `ErrorsReachedThreshold()` function will return false at next tick and reset backoff immediately because we cannot get error from the changefeed in such short period of time. In other words, adding `deltaBackoffInterval` gives an chance to capture errors.